carbons/carbons.go

```diff
@@ -110,3 +111,40 @@ func Unwrap(del *delay.Delay, r xml.TokenReader) (xml.TokenReader, xml.StartElem
 	out, err := forward.Unwrap(del, xmlstream.Inner(r))
 	return out, se, err
 }
+
+// Private is an xmlstream.Transformer that excludes a <message/> from being forwarded to
+// other Carbons-enabled resources, by adding a <private/> element and a <no-copy/> hint.
+// The first call to consume the output stream will return an error if the input stream
+// is not a message element with an appropriate namespace.
+func Private(r xml.TokenReader) xml.TokenReader {
```

Comment:
The whole body of this may already exist as [`xmlstream.InsertFunc`](https://pkg.go.dev/mellium.im/xmlstream#InsertFunc), it might be worth looking into whether a global, internal, call to InsertFunc can replace the body of this function here. As a plus, this would give us the ability to make sure it only applies to messages at level 0 (or 1? I forget if it counts 1 indexed or 0 indexed off the top of my head).